README.md

```diff
@@ -30,165 +32,471 @@ The two guiding principles are:
+
+### Options
 
-#### `hops-redux`
+There is another way to pass configuration to your Hops application, which is being used for runtime values:
+
+**`src/app.js`**
+
+```javascript
+const reducers = {
+  increment(state = 0, action) {
+    return action.type === 'INCREMENT' ? state + action.payload : state;
+  },
+};
+
+export default render(<MyApp />, { redux: { reducers } });
+```
+
+## Presets
+
+Almost everything in Hops is a preset that just needs to be installed / configured in order to start using it.
+
+### Installing presets
+
+In order to install a preset you need to add it as a dependency to your application:
+
+```shell
+$ npm install --save hops-redux
+```
+
+Sometimes presets have `peerDependencies` which need to be installed as well - take a look at the [indiviual preset sections](#available-presets) or watch out for peer dependency warnings in your terminal.
+
+### Activating presets
+
+By default Hops will automatically find all explicitly installed presets by looking through the top-level dependencies in your `package.json`.
+
+If you prefer, you can also explicitly list the presets that you want to use under the `"presets"` key in your [settings](#settings). This will disable automatic discovery of Hops presets.
+
+**`package.json`**
+
+```json
+{
+  "hops": {
+    "presets": ["hops", "hops-redux"]
+  }
+}
+```
+
+Some presets require (or allow) additional configuration. Read the sections below for each of the presets you are using to find out what settings and options are available to you.
+
+### Available presets
+
+#### [`hops`](https://github.com/xing/hops/tree/master/packages/cli)
+
+This is the default preset that contains the basic building blocks for Hops itself, therefore it will always be available when creating a Hops application. It takes care of setting up the CLI, Express.js server, webpack and React.js support.
+
+##### Components
+
+###### `<Miss />`
+
+###### `<Status code={Number} />`
+
+###### `<Header name={String} value={String} />`
+
+###### `Import(module, [exportName])`
+
+###### `withServerData(Component)`
+
+###### `withConfig(Component)`
+
+#### [`hops-redux`](https://github.com/xing/hops/tree/master/packages/redux)
 
 This preset will set-up a Redux store, take care of dehydration / rehydration and wrapping your application in a `<Provider />`.
 
-Install it to your project:
+Install it and its peer dependencies to your project:
 
-```bash
-$ yarn add hops-redux@next@next react-redux redux redux-thunk
+```shell
+npm install --save hops-redux react-redux redux redux-thunk
 ```
 
-And pass your reducers as [render options](https://github.com/xing/hops/tree/master/packages/redux#render-options):
+And pass your reducers as [options](https://github.com/xing/hops/tree/master/packages/redux#render-options) to the `render()` function:
+
+**`src/app.js`**
 
 ```javascript
 import { render } from 'hops-react';
```

Comment:
Should this really be 'hops-react' or just 'hops'? There are more code examples below that also import from 'hops-react'.

Reply:
done